src/pybind/mgr/volumes/module.py

```diff
@@ -137,6 +137,24 @@ class Module(orchestrator.OrchestratorClientMixin, MgrModule):
                     "and optionally, in a specific subvolume group",
             'perm': 'rw'
         },
+        {
+            'cmd': 'fs subvolume extend '
+                   'name=vol_name,type=CephString '
+                   'name=sub_name,type=CephString '
+                   'name=newsize,type=CephInt,req=true '
+                   'name=group_name,type=CephString,req=false ',
+            'desc': "Extend a CephFS subvolume",
+            'perm': 'rw'
+        },
+        {
+            'cmd': 'fs subvolume shrink '
+                   'name=vol_name,type=CephString '
+                   'name=sub_name,type=CephString '
+                   'name=newsize,type=CephInt,req=true '
+                   'name=group_name,type=CephString,req=false ',
+            'desc': "Shrink a CephFS subvolume",
+            'perm': 'rw'
+        },
```

Comment:
Why do we need two different commands and not just `fs subvolume resize`? The distinction doesn't seem important to me.

Reply:
@ajarr I remember we had a discussion on this earlier. Can you provide the reason?

Reply:
Guess, I just wanted to replicate the manila extend/shrink APIs.

I checked with Shyam, and discussed Ceph CSI requirements. Looks like `fs subvolume resize` is the interface that would work better for the CSI use case too.
So with `resize` we would just set the new desired quota, irrespective of the existing or the used size of the subvolume.

We were wondering if it would be useful to add two non-mandatory boolean options:

*  `--expand-only` (by default false), when set to true, the `resize` interface would check whether the desired size > original size (quota value already set). If not, error out with EINVAL. 

*  `--shrink-only` (by default false), when set to true, check whether the desired size < original size. If not, error out with EINVAL.

Reply:
The options seem harmless enough as long as they have tests.

Reply:
@ajarr As I have converted it to a `resize` command now, I think we should make those checks (mentioned above) mandatory and not optional. As I understand, we don't want the user to give a resize value which is lesser than the current used size. That is the only restriction I could see for a `resize` command. This way it looks better than leaving a loophole when the user sets `--expand-only=false` and `--shrink-only=false`. The user can resize the subvolume to any size above the used size. (except is there a max resize limit, which we need to consider?).

Let me know your thoughts.

Reply:
It's better to have `expand-only` and `shrink-only` false by default.

It's fine to give a resize value that is less than the current used size. We should just log a warning with other details as mentioned in https://github.com/ceph/ceph/pull/30054#discussion_r331742572

There is no max resize limit.